data_steward/analytics/cdr_ops/rt_cdr_qc/cdr_deid_qa_report3_col_suppression.py

```diff
@@ -178,85 +155,62 @@
 SUM(CASE WHEN provider_id IS NOT NULL THEN 1 ELSE 0 END) AS n_provider_id_not_null
 FROM `{{project_id}}.{{deid_cdr}}.condition_occurrence`
 """)
-q = query.render(project_id=project_id, deid_cdr=deid_cdr)
-df1 = execute(client, q)
-if df1.loc[0].sum() == 0:
-    df = df.append({
-        'query': 'Query5 Condition provider_id',
-        'result': 'PASS'
-    },
-                   ignore_index=True)
+q = query.render(project_id=project_id,deid_cdr=deid_cdr)  
+df1=execute(client, q) 
+if df1.loc[0].sum()==0:
+ df = df.append({'query' : 'Query5 Condition provider_id', 'result' : 'PASS'},  
+                ignore_index = True) 
 else:
-    df = df.append(
-        {
-            'query': 'Query5 Condition provider_id',
-            'result': 'Failure'
-        },
-        ignore_index=True)
-df1
+ df = df.append({'query' : 'Query5 Condition provider_id', 'result' :  'Failure'},  
+                ignore_index = True) 
+df1        
 
-# # 6 Verify the following columns in the DEVICE_EXPOSURE table have been set to null:
+# # 6 Verify the provider_id in the DEVICE_EXPOSURE table is null and visit_detail_id is not all null:
 #
-# <font color='red'> Uncomment below when visit_detail_id is available
+# <font color='red'> DST-1423,updates, only provide_id is null.
 
 # +
 # query6 has error
 # no this column called visit_detail_id?
-
+    
 query = JINJA_ENV.from_string("""
 SELECT
 SUM(CASE WHEN provider_id IS NOT NULL THEN 1 ELSE 0 END) AS n_provider_id_not_null,
 SUM(CASE WHEN visit_detail_id IS NOT NULL THEN 1 ELSE 0 END) AS n_visit_detail_id_not_null
 
 FROM `{{project_id}}.{{deid_cdr}}.device_exposure`
 """)
-q = query.render(project_id=project_id, deid_cdr=deid_cdr)
-df1 = execute(client, q)
-
-if df1.loc[0].sum() == 0:
-    df = df.append({
-        'query': 'Query6 provider_id in Device',
-        'result': 'PASS'
-    },
-                   ignore_index=True)
+q = query.render(project_id=project_id,deid_cdr=deid_cdr)  
+df1=execute(client, q) 
+
+if df1.loc[0].sum()==0 or df1.iat[0,0] !=0:
+ df = df.append({'query' : 'Query6 provider_id in Device', 'result' : 'Failure'},  
+                ignore_index = True) 
 else:
-    df = df.append(
-        {
-            'query': 'Query6 provider_id in Device',
-            'result': 'Failure'
-        },
-        ignore_index=True)
+ df = df.append({'query' : 'Query6 provider_id in Device', 'result' :  'Pass'},  
+                ignore_index = True) 
 df1
 # -
 
-# # 7 Verify the following columns in the DRUG_EXPOSURE table have been set to null:
-# provider_id
+# # 7 Verify the provider_id in the DRUG_EXPOSURE table is null and visit_detail_id is not all null:
 #
-# <font color='red'> no visit_detail_id in old CDR?
-#
-#  Uncomment below when visit_detail_id is available in 2022 CDR
+# <font color='red'> DST-1423,updates, only provide_id is null.
```

Comment:
A similar comment as the one mentioned above would make the results easier to understand.